docs/en_US/reference/experiment_config.rst

```diff
@@ -0,0 +1,575 @@
+===========================
+
+This is the detailed list of experiment config fields.
+For quick start guide, reference the tutorial instead. [TODO]
+
+Notes
+=====
+
+1. This document list field names as separated words.
+   They should be spelled in ``snake_case`` for Python library ``nni.experiment``, and are normally spelled in ``camelCase`` for YAML files.
+
+2. In this document type of fields are expressed in `Python type hint <https://docs.python.org/3/library/typing.html>`_ format.
+   Therefore JSON objects are called `dict` and arrays are called `list`.
+
+.. _Path:
+.. _directory:
+
+3. Some fields take a path to file or directory.
+   Unless otherwise noted, both absolute path and relative path are supported, and ``~`` can be used for home directory.
+
+   - When written in YAML file, relative paths are relative to the directory containing that file.
+   - When assigned in Python code, relative paths are relative to current working directory.
+   - All relative paths are converted to absolute when loading YAML file into Python class, and when saving Python class to YAML file.
+
+4. Setting a field to ``None`` or ``null`` is equivalent to not setting the field.
+
+ExperimentConfig
+================
+
+experiment name
+---------------
+
+Mnemonic name of the experiment. This will be shown in web UI and nnictl.
+
+type: ``Optional[str]``
+
+
+search space file
+-----------------
+
+Path_ to a JSON file containing the search space.
+
+type: ``Optional[str]``
+
+Search space format is determined by tuner. Common format for built-in tuners is documeted `here <https://nni.readthedocs.io/en/stable/Tutorial/SearchSpaceSpec.html>`_.
+
+Mutually exclusive to `search space`_.
+
+
+search space
+------------
+
+Search space object.
+
+type: ``Optional[Any]``
+
+The format is determined by tuner. Common format for built-in tuners is documented here `here <https://nni.readthedocs.io/en/stable/Tutorial/SearchSpaceSpec.html>`_.
+
+Note that ``None`` means "no such field" so empty search space should be written as ``{}``.
+
+Mutually exclusive to `search space file`_.
+
+
+trial command
+-------------
+
+Command(s) to launch trial.
+
+type: ``str``
+
+Bash will be used on Linux and macOS. PowerShell will be used on Windows.
+
+
+trial code directory
+--------------------
+
+`Path`_ to the directory containing trial source files.
+
+type: ``str``
+
+default: ``"."``
+
+All files in this directory will be sent to training machine, unless there is a ``.nniignore`` file [TODO:link]
+
+
+trial concurrency
+-----------------
+
+Specify how many trials should be run concurrently.
+
+type: ``int``
+
+The real concurrency also depends on hardware resources and may be less than this value.
+
+
+trial gpu number
+----------------
+
+Number of GPUs used by each trial.
+
+type: ``Optional[int]``
+
+If set to zero, trials will have no access to any GPU. 
+
+If not specified, trials will be created and scheduled as if they do not use GPU,
```

Comment:
What is the conclusion from last meeting? What does -1, 0, None mean respectively?

Reply:
No change from v1.